federation/pkg/federation-controller/service/dns.go

```diff
@@ -164,86 +166,90 @@ func (s *ServiceController) ensureDnsRrsets(dnsZoneName, dnsName string, endpoin
 		return err
 	}
 	if rrset == nil {
-		if endpointReachable {
-			// It doesn't exist yet, so create it, if we indeed have healthy endpoints
-			if len(endpoints) < 1 {
-				// There are no endpoint addresses at this level, so CNAME to uplevel, if provided
-				if uplevelCname != "" {
-					newRrset := rrsets.New(dnsName, []string{uplevelCname}, minDnsTtl, rrstype.CNAME)
-					rrset, err = rrsets.Add(newRrset)
-					if err != nil {
-						return err
-					}
-				}
-				// else we want no record, and we have no record, so we're all good.
-			} else {
-				// We have valid endpoint addresses, so just add them as A records.
-				// But first resolve DNS names, as some cloud providers (like AWS) expose
-				// load balancers behind DNS names, not IP addresses.
-				resolvedEndpoints, err := getResolvedEndpoints(endpoints)
-				if err != nil {
-					return err // TODO: We could potentially add the ones we did get back, even if some of them failed to resolve.
-				}
-				newRrset := rrsets.New(dnsName, resolvedEndpoints, minDnsTtl, rrstype.A)
+		glog.V(4).Infof("No recordsets found for DNS name %q.  Need to add either A records (if we have healthy endpoints), or a CNAME record to %q", dnsName, uplevelCname)
+		if len(endpoints) < 1 {
+			glog.V(4).Infof("There are no healthy endpoint addresses at level %q, so CNAME to %q, if provided", dnsName, uplevelCname)
+			if uplevelCname != "" {
+				glog.V(4).Infof("Creating CNAME to %q for %q", uplevelCname, dnsName)
+				newRrset := rrsets.New(dnsName, []string{uplevelCname}, minDnsTtl, rrstype.CNAME)
 				rrset, err = rrsets.Add(newRrset)
```

Comment:
Add `glog.V(4).Infof("Adding recordset %v", newRrset)` as you do in the else block.